app/controllers/api/storage.php

```diff
@@ -67,6 +67,11 @@
         /** @var Appwrite\Event\Event $audits */
         /** @var Appwrite\Stats\Stats $usage */
 
+        // Maximum allowed size validation
+        if ($maximumFileSize > (int) App::getEnv('_APP_STORAGE_LIMIT', 0)) {
+            throw new Exception('Bucket maximum file size is larger than _APP_STORAGE_LIMIT.', 400, Exception::ATTRIBUTE_VALUE_INVALID);
+        }
+
```

Comment:
Why aren't we using the range validator to do this for us like we do in functions ? 
https://github.com/appwrite/appwrite/blob/5ab38c0f9889e005b012ea267488d030f7002396/app/controllers/api/functions.php#L52